README.md

```diff
@@ -7,94 +7,201 @@ A third party tool which enables an author to record coding tasks by using an in
-
-    You can get rid of the above mentioned error by installing the following package
-
-        sudo apt-get install libasound2-dev
-
-    After that, run `npm install` command again and if you see any error something like that is given below, just ignore it and proceed to next step
-
-        gyp ERR! build error
-        gyp ERR! stack Error: `make` failed with exit code: 2
-        gyp ERR! stack     at ChildProcess.onExit (/home/umar/.nvm/versions/node/v12.10.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:196:23)
-        gyp ERR! stack     at ChildProcess.emit (events.js:209:13)
-        gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
-        gyp ERR! System Linux 4.15.0-20-generic
-        gyp ERR! command "/home/umar/.nvm/versions/node/v12.10.0/bin/node" "/home/umar/.nvm/versions/node/v12.10.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
-        gyp ERR! cwd /home/umar/Documents/Work/codecast/node_modules/speaker
-        gyp ERR! node -v v12.10.0
-        gyp ERR! node-gyp -v v5.0.3
-        gyp ERR! not ok
-
-
-5. To compile assets, run
-
-        npm run build
-
-    You may get the above mentioned error again, but you can ignore it and run the next command
-
-        npm run start
-
-6.  Go to `http://localhost:8001/login` link and login using your admin credentials
-
-        Admin Email: <YOUR ADMIN EMAIL ADDRESS THAT YOU STORED IN DATABASE>
-        Password: admin
+## **Update Distribution**
+
+Make sure to update distribution:
+```console
+# apt-get update && apt-get dist-upgrade
+```
+
+## **Create a User for CodeCast**
+Create a user to run CodeCast:
+```console
+# adduser codecast
+```
+
+## **Install Requirements**
+Install NodeJS as follows:
+```console
+# curl -sL https://deb.nodesource.com/setup_12.x | bash -
+# apt-get install -y nodejs build-essential libasound2-dev mysql-server
+```
+
+## **Create CodeCast Database**
+Create database as follows:
+```sql
+CREATE DATABASE codecast;
+GRANT ALL ON codecast.* TO 'codecast'@'localhost' IDENTIFIED BY 'password';
+FLUSH PRIVILEGES;
+```
+
+## **Create NodeJS Environment for `codecast` User**
+
+Switch to codecast user to perform further setup steps.
+
+Add the following environment variables to the end of `~/.bashrc`:
+```bash
+# set NPM directory
+NPM_DIR="$HOME/.npm"
+
+# set PATH to include local NPM directory
+PATH="$NPM_DIR/bin:$PATH"
+
+# Unset manpath so we can inherit from /etc/manpath via the `manpath` command
+unset MANPATH  # delete if you already modified MANPATH elsewhere in your configuration
+MANPATH="$NPM_DIR/share/man:$(manpath)"
+
+# Tell Node about these packages
+NODE_PATH="$NPM_DIR/lib/node_modules:$NODE_PATH"
+```
+
+Create `~/.npmrc` with following content:
+```text
+prefix = ${HOME}/.npm
+```
+
+For the above environment changes to be effective:
+```console
+source ~/.bashrc
+```
+
+## **Setup CodeCast**
```

Comment:
FYI: I think we have skipped Codecast's installation here e.g: `npm install, npm run build, etc`, will add that in a new commit